source/_drafts/videojs-contrib-ads-6.md

```diff
@@ -0,0 +1,41 @@
+---
+title: videojs-contrib-ads 6
+author:
+  name: Greg Smith
+  github: incompl
+tags:
+---
+
+A major update to videojs-contrib-ads has been released. In this post, we'll take a look at what videojs-contrib-ads offers and what has improved in this latest version.
+
+## What is contrib-ads?
+
+Videojs-contrib-ads is a framework for creating Video.js ad plugins. Seamlessly integrating ad support into a video player can be a daunting task, especially if you have other plugins that may be effected. For example, contrib-ads has a feature called [Redispatch](https://github.com/videojs/videojs-contrib-ads#redispatch) that makes sure [media events](https://developer.mozilla.org/en-US/docs/Web/Guide/Events/Media_events) are not impacted by the presence of ads. You may have some JavaScript on your site, or another Video.js plugin, that responds to events such as the `play` event. With contrib-ads, if an ad plays, it will not send an additional `play` event. This keeps things simple as possible without breaking other plugins. [Additional benefits of using contrib-ads are listed in the readme](https://github.com/videojs/videojs-contrib-ads#benefits).
+
+## Version 6
+
+Version 6 of contrib-ads is a maintenance release that includes a major code refactor. The core of the project's behavior is handled by a state machine that advances through various states as the player plays content, preroll ads, midroll ads, and postroll ads. The initial state machine was designed to implement a specific set of functionality when the project was first conceptualized. Over the years, the project became much more fully-featured and many new scenarios and edge-cases arose. In this refactor, the state machine was updated to precisely match the modern feature-set. This has resulted in a more maintainable and reliable codebase.
+
+## How it works
+
+For a history of how the state machine has evolved over time, [there is detailed information in this Github issue](https://github.com/videojs/videojs-contrib-ads/issues/320). In this article we're going to focus on how it works in version 6.
+
+Here is a diagram of the current states:
+
+![](./ad-states.png)
```

Comment:
no link?

Reply:
That is where alt text goes, not a link, if I understand the question correctly.

Reply:
our template will put this text as a title for the image as well:
```md
![IE8](IE8-small.png)
```
![screen shot 2018-03-09 at 17 17 34](https://user-images.githubusercontent.com/535884/37232610-cae4ce98-23bd-11e8-829e-792336b5534b.png)